src/index.js

```diff
@@ -71,91 +72,125 @@ jsonSchemaAvro._hasEnum = (schema) => {
 
 jsonSchemaAvro._isRequired = (list, item) => list.includes(item)
 
-jsonSchemaAvro._convertProperties = (schema = {}, required = []) => {
+jsonSchemaAvro._getUniqueName = (name, type, namesCache = {}) => {
```

Comment:
Could we prefix all names with their parents names to ensure they are unique. `somefield1`, `somefield2` etc. does not relate well to the input as it assumes ordering